### PR TITLE
productsubscription: move token generation to licensing, change token prefix

### PR DESCRIPTION
**Pull request by bobheadxi**
With #52634, we are making the product subscription access token explicitly the license key, rather than hiding it as an implementation detail. This change moves access token generation to the `licensing` package so we can generate it from anywhere, and creates a new token prefix `slk_` ("S"ourcegraph "L"icense "K"ey) to make the relationship explicit. We retain support for the old prefix for now.

## Test plan

Tests. Added a new test to cover back-compat with the old format as well.

### CODE REVIEW

File: enterprise/cmd/cody-gateway/internal/actor/productsubscription/productsubscription.go [hunk 1]
```diff
@@ -149,7 +153,7 @@ func (s *Source) checkAccessToken(ctx context.Context, token string) (*dotcom.Ch
 
 	for _, gqlerr := range gqlerrs {
 		if gqlerr.Extensions != nil && gqlerr.Extensions["code"] == codygateway.GQLErrCodeProductSubscriptionNotFound {
-			return nil, actor.ErrAccessTokenDenied{"associated product subscription not found"}
+			return nil, actor.ErrAccessTokenDenied{Reason: "associated product subscription not found"}
```
Comment (bobheadxi) [inline enterprise/cmd/cody-gateway/internal/actor/productsubscription/productsubscription.go:156]: Linter complaint

File: enterprise/internal/licensing/access_token.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"encoding/hex"
+
+	"github.com/sourcegraph/sourcegraph/internal/hashutil"
+)
+
+// LicenseKeyBasedAccessTokenPrefix is the prefix used for identifying tokens
+// generated for product subscriptions based on a Sourcegraph license key.
+const LicenseKeyBasedAccessTokenPrefix = "slk_" // "(S)ourcegraph (L)icense (K)ey"
```
Comment (eseliger) [inline enterprise/internal/licensing/access_token.go:11]: what I thought when I read SLK 😆  image

**Review by eseliger — APPROVED** [review]

File: enterprise/internal/licensing/access_token.go (hunk 2 above)
Comment (bobheadxi) [reply enterprise/internal/licensing/access_token.go:11]: official mascot? 😁

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #52811** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/52811" target="_blank">Graphite</a>  👈
    * **PR #52812** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/52812" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/52811?utm_source=stack-comment).

**Pull request merged**